rllib/utils/metrics/stats.py

```diff
@@ -14,6 +15,43 @@
 torch, _ = try_import_torch()
 
 
+@DeveloperAPI
+def compute_percentiles(sorted_list, percentiles):
+    """Compute percentiles from an already sorted list.
+
+    Note that this will not raise an error if the list is not sorted to avoid overhead.
+
+    Args:
+        sorted_list: A list of numbers sorted in ascending order
+        percentiles: A list of percentile values (0-100)
+
+    Returns:
+        A dictionary mapping percentile values to their corresponding data values
+    """
+    n = len(sorted_list)
+
+    if n == 0:
+        return {p: None for p in percentiles}
+
+    results = {}
+
+    for p in percentiles:
+        index = (p / 100) * (n - 1)
+
+        if index.is_integer():
```

Comment:
Dumb question: why not using here `numpy.percentile`?

Reply:
`numpy.percentiles` does not assume sorted values. So it will end up sorting with at best n log n.
So we could just not sort values when aggregating and us np.percentiles in the end.
But if we train at scale and have 1000 parallel env runners, it will have a pretty sizeable impact if we sort on env runners first so that we can use heapq to merge the already sorted lists in linear time.